### PR TITLE
feat: implement cloud{provider,platform}, aws.{ecs.*,log.*} in ECS detector

### PR DESCRIPTION
**Pull request by mmanciop**
This PR is the look-alike of others like https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1083, https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1212 and https://github.com/open-telemetry/opentelemetry-go-contrib/pull/2626.

### CODE REVIEW

**Review by brettmc — CHANGES_REQUESTED** [review]
Welcome and thanks for your contribution. This looks good, just a couple of small issues/improvements from me.

File: src/Aws/src/Ecs/Detector.php [hunk 1]
```diff
@@ -39,34 +42,48 @@ class Detector implements ResourceDetectorInterface
...
+        $this->requestFactory = $requestFactory;
     }
 
     /**
-     * If running on ECS, runs getContainerId(), getClusterName(), and
-     * returns resource with valid extracted values
-     * If not running on ECS, returns empty rsource
+     * If running on ECS with an ECS agent v1.3, returns a resource with the following attributes set:
+     *
+     * -
```
[the remark this answers is not in the document]
Comment (mmanciop) [reply src/Aws/src/Ecs/Detector.php]: Yeah they remained in the keyboard :-)

Done in 8260f9a

File: src/Aws/src/Ecs/Detector.php [hunk 2]
```diff
@@ -39,34 +42,48 @@ class Detector implements ResourceDetectorInterface
...
+     *
+     * If not running on ECS, returns empty resource.
      */
     public function getResource(): ResourceInfo
     {
+        $metadataEndpointV4 = getenv(self::ECS_METADATA_KEY_V4);
         // Check if running on ECS by looking for below environment variables
-        if (!getenv(self::ECS_METADATA_KEY_V4) && !getenv(self::ECS_METADATA_KEY_V3)) {
+        if (!$metadataEndpointV4 && !getenv(self::ECS_METADATA_KEY_V3)) {
             // TODO: add 'Process is not running on ECS' when logs are added
```
[the remark this answers is not in the document]
Comment (mmanciop) [reply src/Aws/src/Ecs/Detector.php]: Done in 94c7d9a59131319f915b18fe9db92654648169ea

**Review by brettmc — APPROVED** [review]

**Comment by mmanciop** [comment]
@brettmc what needs to happen before we can merge this?

**Comment by bobstrecansky** [comment]
@mmanciop - merging now! :)

**Pull request merged**